R/java_env.R

```diff
@@ -267,25 +274,27 @@ java_check_version_cmd <- function(
   }
 
   # Check Java path and version using system commands
-  success <- java_check_version_system()
+  major_java_version <- java_check_version_system(quiet = quiet)
 
   # restore original JAVA_HOME that was in the environment before the function was called
   if (!is.null(java_home)) {
     Sys.setenv(JAVA_HOME = old_java_home)
   }
 
-  return(success)
+  return(major_java_version)
 }
 
-#' Check and print Java path and version
+#' Check and print Java path and version using system commands
 #'
 #' This function checks the Java executable path and retrieves the Java version,
 #' then prints these details to the console.
-#'
+#' @inheritParams java_check_version_cmd
+#' @return A `character` vector of length 1 containing the major Java version.
 #' @keywords internal
-#'
-#' @return TRUE if successful, otherwise FALSE.
-java_check_version_system <- function() {
+#' 
+java_check_version_system <- function(
+  quiet
+) {
   which_java <- tryCatch(
```

Comment:
Is there a reason you don't use `Sys.which()` here?

Reply:
I honestly did not know that `Sys.which()` existed (just like many other good base R functions). Will happily fix.